include/iotc_error.h

```diff
@@ -21,118 +21,105 @@
 extern "C" {
 #endif
 
+/*! \file
+ * @brief Defines state messages and their numeric codes.
+ */
+
 /**
- * @name iotc_state_t
- * @brief Internal error codes.
- *
- * IOTC_STATE_OK represents success. Others represent errors.
+ * @typedef iotc_state_t
+ * @brief The internal state.
```

Comment:
I'd phrase something like this:
The state of the client, returned by most of both API and internal functions. Only a subset of the error codes is perceivable from outside. Majority of the codes occurs only internally.

Reply:
Done